app/src/main/java/com/aleksandra/go4mytrip/TripAdapter.kt

```diff
@@ -0,0 +1,60 @@
+package com.aleksandra.go4mytrip
+
+import android.content.Context
+import android.content.Intent
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageButton
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+
+class TripAdapter(private val context: Context, private val mData: MutableList<TripModel?>?, private val tripsListener: TripsListener) : RecyclerView.Adapter<TripAdapter.MyViewHolder>() {
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
+        val view: View
+        val mInflater = LayoutInflater.from(context)
+        view = mInflater.inflate(R.layout.item_trip, parent, false)
+        return MyViewHolder(view)
+    }
+
+    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
+        holder.textViewTitle.text = mData!![position]!!.title
+        holder.imgTrip.setImageResource(mData[position]!!.imageTrip)
+        holder.dateStart.text = mData[position]!!.tripDate
+        holder.dateEnd.text = mData[position]!!.tripDateEnd
+        holder.imgTrip.setOnClickListener {
+            val id = mData[position]!!.tripId
+            val title = mData[position]!!.title
+            val image = mData[position]!!.imageTrip
+            val dateS = mData[position]!!.tripDate
+            val dateE = mData[position]!!.tripDateEnd
+            val timeStart = mData[position]!!.timeStart
+            val timeEnd = mData[position]!!.timeEnd
+            val intent = Intent(context, DetailTrip::class.java)
+            intent.putExtra("idTripToDetail", id)
+            intent.putExtra("title", title)
```

Comment:
replace these fields with a dedicated model that implements the Parcelable interface